plugins/system/userlogs/userlogs.php

```diff
@@ -129,474 +171,538 @@ public function onContentAfterSave($context, $article, $isNew)
+						ucfirst(JText::_($type)), $message_to_array['title']
+					);
+				}
+				elseif ($message_to_array['value'] == 2)
+				{
+					$message = JText::sprintf('PLG_SYSTEM_USERLOG_ON_CONTENT_CHANGE_STATE_ARCHIVED_MESSAGE',
+						ucfirst(JText::_($type)), $message_to_array['title'], $message_to_array['title']
+					);
+				}
+				elseif ($message_to_array['value'] == -2)
+				{
+					$message = JText::sprintf('PLG_SYSTEM_USERLOG_ON_CONTENT_CHANGE_STATE_TRASHED_MESSAGE',
+						ucfirst(JText::_($type)), $message_to_array['title']
+					);
+				}
+				break;
+			case 'onExtensionAfterInstall':
+				$message = JText::sprintf('PLG_SYSTEM_USERLOG_ON_EXTENSION_AFTER_INSTALL_MESSAGE', $message_to_array['extenstion_name']);
+				break;
+			case 'onExtensionAfterUninstall':
+				$message = JText::sprintf('PLG_SYSTEM_USERLOG_ON_EXTENSION_AFTER_UNINSTALL_MESSAGE', $message_to_array['extenstion_name']);
+				break;
+			case 'onExtensionAfterUpdate':
+				$message = JText::sprintf('PLG_SYSTEM_USERLOG_ON_EXTENSION_AFTER_UPDATE_MESSAGE', $message_to_array['extenstion_name']);
+				break;
+			case 'onUserAfterSave':
+				if ($message_to_array['isNew'] == 'false')
+				{
+					$message = JText::sprintf('PLG_SYSTEM_USERLOG_ON_USER_AFTER_SAVE_MESSAGE', $message_to_array['edited_user']);
+				}
+				else
+				{
+					$message = JText::sprintf('PLG_SYSTEM_USERLOG_ON_USER_AFTER_SAVE_NEW_MESSAGE', $message_to_array['edited_user']);
+				}
+				break;
+			case 'onUserAfterDelete':
+				$message = JText::sprintf('PLG_SYSTEM_USERLOG_ON_USER_AFTER_DELETE_MESSAGE', $message_to_array['edited_user']);
+				break;
+			case 'onUserAfterSaveGroup':
+				if ($message_to_array['isNew'] == 'false')
+				{
+					$message = JText::sprintf('PLG_SYSTEM_USERLOG_ON_USER_AFTER_SAVE_GROUP_MESSAGE', $message_to_array['title']);
+				}
+				else
+				{
+					$message = JText::sprintf('PLG_SYSTEM_USERLOG_ON_USER_AFTER_SAVE_GROUP_NEW_MESSAGE', $message_to_array['title']);
+				}
+				break;
+			case 'onUserAfterDeleteGroup':
+				$message = JText::sprintf('PLG_SYSTEM_USERLOG_ON_USER_AFTER_DELETE_GROUP_MESSAGE', $message_to_array['deleted_group']);
+				break;
+			case 'onExtensionAfterSave':
+				if ($message_to_array['isNew'] == 'false')
+				{
+					$message = JText::sprintf('PLG_SYSTEM_USERLOG_ON_EXTENSION_AFTER_SAVE_MESSAGE', ucfirst(JText::_($type)));
+				}
+				else
+				{
+					$message = JText::sprintf('PLG_SYSTEM_USERLOG_ON_EXTENSION_AFTER_SAVE_NEW_MESSAGE', JText::_($type));
+				}
+
+				if (!empty($message_to_array['title']))
+				{
+					$message = $message . JText::sprintf('PLG_SYSTEM_USERLOG_TITLED', $message_to_array['title']);
+				}
+				break;
+			case 'onExtensionAfterDelete':
+				$message = JText::sprintf('PLG_SYSTEM_USERLOG_ON_EXTENSION_AFTER_DELETE_MESSAGE', $extension);
+
+				if (!empty($message_to_array['title']))
+				{
+					$message = $message . JText::sprintf('PLG_SYSTEM_USERLOG_TITLED', $message_to_array['title']);
+				}
+				break;
```

Comment:
That whole block looks, let me say it nice, horrific ;) Any change to improve that? :-)